File: Segundo/Segundo_Cuatrimestre/Informatica/PDOO/SUBIR/Examen19-20/WeaponType.rb
#encoding: UTF-8

module Deepspace
	module WeaponType
		class Type
			def initialize (p) # p es float
				@power = p
			end
			
			attr_reader	:power
			
			def to_s
				if (self == WeaponType::LASER)
					type = "LASER"
				else 
					if (self == WeaponType::MISSILE)
						type = "MISSILE"
					else
						type = "PLASMA"
					end
				end
				
				return type
			end
		end
		
		LASER = Type.new(2.0)
		MISSILE = Type.new(3.0)
		PLASMA = Type.new(4.0)
		
	end

end

File: Segundo/Segundo_Cuatrimestre/Informatica/PDOO/SUBIR/MARCIANITOS_P3/Marcianitos/src/deepspace/WeaponType.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Enum.java to edit this template
 */

package deepspace;

/**
 *
 * @author mario
 */
public enum WeaponType {
	LASER(2.0f), MISSILE(3.0f), PLASMA(4.0f);
    
    private final float power;
    
    WeaponType (float the_power) {
        this.power = the_power;
        
    }
    
    float getPower(){
        return (this.power);
    }
}
